pkg/setting/setting_unified_alerting.go

```diff
@@ -103,31 +103,40 @@ func (cfg *Cfg) readUnifiedAlertingEnabledSetting(section *ini.Section) (*bool,
 	// At present an invalid value is considered the same as no value. This means that a
 	// spelling mistake in the string "false" could enable unified alerting rather
 	// than disable it. This issue can be found here
-	unifiedAlerting, err := section.Key("enabled").Bool()
-	if err != nil {
+	hasEnabled := section.Key("enabled").Value() != ""
+	if !hasEnabled {
 		// TODO: Remove in Grafana v9
 		if cfg.IsFeatureToggleEnabled("ngalert") {
 			cfg.Logger.Warn("ngalert feature flag is deprecated: use unified alerting enabled setting instead")
 			// feature flag overrides the legacy alerting setting
 			legacyAlerting := false
 			AlertingEnabled = &legacyAlerting
-			unifiedAlerting = true
+			unifiedAlerting := true
 			return &unifiedAlerting, nil
 		}
 
 		// if legacy alerting has not been configured then enable unified alerting
 		if AlertingEnabled == nil {
-			unifiedAlerting = true
+			unifiedAlerting := true
 			return &unifiedAlerting, nil
 		}
 
 		// enable unified alerting and disable legacy alerting
 		legacyAlerting := false
 		AlertingEnabled = &legacyAlerting
-		unifiedAlerting = true
+		unifiedAlerting := true
 		return &unifiedAlerting, nil
 	}
 
+	unifiedAlerting, err := section.Key("enabled").Bool()
+	if err != nil {
+		// the value for unified alerting is invalid so disable all alerting
+		legacyAlerting := false
+		AlertingEnabled = &legacyAlerting
+
+		return nil, errors.New("the value for enabled must be true or false")
```

Comment:
I think we should be more detailed here and point out where exactly the problem is. Something like "setting `unified_alerting.enabled` has invalid value %s. Supported only: [true, false]"